pkg/cmd/cli/cmd/logs.go

```diff
@@ -0,0 +1,176 @@
+package cmd
+
+import (
+	"fmt"
+	"io"
+	"sort"
+	"strings"
+
+	"github.com/spf13/cobra"
+
+	kclient "k8s.io/kubernetes/pkg/client/unversioned"
+	"k8s.io/kubernetes/pkg/fields"
+	kcmd "k8s.io/kubernetes/pkg/kubectl/cmd"
+	cmdutil "k8s.io/kubernetes/pkg/kubectl/cmd/util"
+	"k8s.io/kubernetes/pkg/labels"
+
+	buildapi "github.com/openshift/origin/pkg/build/api"
+	"github.com/openshift/origin/pkg/client"
+	"github.com/openshift/origin/pkg/cmd/util/clientcmd"
+)
+
+const (
+	logsLong = `
+Print the logs for a container in a pod
+
+If the pod has only one container, the container name is optional.`
+
+	logsExample = `  # Returns snapshot of ruby-container logs from pod 123456-7890.
+  $ %[1]s logs 123456-7890 -c ruby-container
+
+  # Starts streaming of ruby-container logs from pod 123456-7890.
+  $ %[1]s logs -f 123456-7890 -c ruby-container`
+)
+
+type OpenShiftLogsOptions struct {
+	KubeClient   *kclient.Client
+	OriginClient *client.Client
+
+	Namespace      string
+	ResourceString string
+	ContainerName  string
+	Follow         bool
+	Interactive    bool
+	Previous       bool
+	Out            io.Writer
+}
+
+// NewCmdLogs creates a new pod log command
+func NewCmdLogs(fullName string, f *clientcmd.Factory, out io.Writer) *cobra.Command {
+	o := &OpenShiftLogsOptions{
+		Out:         out,
+		Interactive: true,
+	}
+
+	cmd := &cobra.Command{
+		Use:     "logs [-f] RESOURCE",
+		Short:   "Print the logs for a container in a pod.",
+		Long:    logsLong,
+		Example: fmt.Sprintf(logsExample, fullName),
+		Run: func(cmd *cobra.Command, args []string) {
+			cmdutil.CheckErr(o.Complete(f, out, cmd, args))
+
+			cmdutil.CheckErr(o.Validate(cmd))
+
+			cmdutil.CheckErr(o.RunLog())
+		},
+		Aliases: []string{"log"},
+	}
+	cmd.Flags().BoolVarP(&o.Follow, "follow", "f", o.Follow, "Specify if the logs should be streamed.")
+	cmd.Flags().BoolVar(&o.Interactive, "interactive", o.Interactive, "If true, prompt the user for input when required. Default true.")
+	cmd.Flags().BoolVarP(&o.Previous, "previous", "p", o.Previous, "If true, print the logs for the previous instance of the container in a pod if it exists.")
+	cmd.Flags().StringVarP(&o.ContainerName, "container", "c", o.ContainerName, "Container name")
+	return cmd
+}
+
+func (o *OpenShiftLogsOptions) Complete(f *clientcmd.Factory, out io.Writer, cmd *cobra.Command, args []string) error {
+	switch len(args) {
+	case 0:
+		return cmdutil.UsageError(cmd, "RESOURCE is required for log")
+
+	case 1:
+		o.ResourceString = args[0]
+	case 2:
+		o.ResourceString = args[0]
+		o.ContainerName = args[1]
+
+	default:
+		return cmdutil.UsageError(cmd, "log RESOURCE [CONTAINER]")
```

Comment:
> -c CONTAINER

I think we want to emphasize the fact that we take more resources, so I've made it match `Use`.